lightning/restore/restore.go

```diff
@@ -747,8 +747,13 @@ func (t *TableRestore) restoreTable(
 		web.BroadcastTableCheckpoint(t.tableName, cp)
 
 		// rebase the allocator so it exceeds the number of rows.
-		cp.AllocBase = mathutil.MaxInt64(cp.AllocBase, t.tableInfo.Core.AutoIncID)
-		t.alloc[0].Rebase(t.tableInfo.ID, cp.AllocBase, false)
+		if t.tableInfo.Core.PKIsHandle && t.tableInfo.Core.ContainsAutoRandomBits() {
+			cp.AllocBase = mathutil.MaxInt64(cp.AllocBase, t.tableInfo.Core.AutoRandID)
+			t.alloc[2].Rebase(t.tableInfo.ID, cp.AllocBase, false)
```

Comment:
the "2" looks pretty magical. could you test the allocator's type instead?

Reply:
should I retain the old logic? only alloc[0] is enough, because all alloc shared the same base in one table

Reply:
no need to retain the old logic.